'use strict';

// Tehtävä 1: Perus Try-Catch käyttö
/*
Kääri koodi try-catch-lohkoon mahdollisten virheiden käsittelyä varten.
Tulosta 'Hups, jotain meni pieleen!' catch-lohkossa konsoliin.
*/

function task1() {
  // Simuloi virhe
  throw new Error('Hups, jotain meni pieleen!');
}

// Tehtävä 2: ReferenceErrorin käsittely
/*
Yritä käyttää määrittelemätöntä muuttujaa try-lohkossa.
Käsittele ReferenceError ja tulosta se konsoliin.
*/

function task2() {
  // Koodi tässä
}

// Tehtävä 3: Finallyn käyttö
/*
Käytä try-catch-finally-lohkoa. Finally-lohkossa tulosta 'Suoritus päättyi'.
*/

function task3() {
  // Koodi tässä
}

// Tehtävä 4: Sisäkkäinen Try-Catch
/*
Sisäkkäisessä try-lohkossa on toinen try-catch-lohko.
Simuloi virhe sisäkkäisessä try-lohkossa ja käsittele se sisäkkäisessä catch-lohkossa.
*/

function task4() {
  // Koodi tässä
}

// Tehtävä 5: Funktio Try-Catchilla
/*
Luo funktio, joka käyttää try-catchia käsittellessään taulukon indeksin ulkopuolista virhettä.
*/

function task5(taulukko, indeksi) {
  // Koodi tässä
}
